### ISSUE TITLE
After freeze websocket client doesnot receive data for a long time and suddenly continues to receive again.

### ISSUE THREAD

**Issue by Fjolnirr** [nasa/trick#1720]
I was playing with the websocket server features of trick and wanted to review the issues about it.  While trying to reproduce the issue #1294 I find another problem that, After freeze websocket client doesnot receive data for a long time and suddenly continues to receive again.

Steps to reproduce:

- Run web server sim(SIM_cannon_analytic edited for websocket usage).
- Connect to sim as websocket client(i used postman) 
- send `var_unpause` command to receive data from websocket server.
- send command below to track a variable.
```
{
    "cmd":"var_add",
    "var_name" : "dyn.cannon.vel[0]"
}
```

- Start the  sim

- You should see flow of messages like this over websocket 
```
 {
    "msg_type": "values",
    "time": 14,
    "values": [
       0
    ]
}
```

- Freeze the sim and websocket stops sending you the response messages(but keeps you connected).
- Start sim again. Still no incoming data to client.(and still connected).
- After waiting for 6 minutes client receives data again. Check the figure below:
![websocket](https://github.com/nasa/trick/assets/40665614/8d884c88-0987-473a-b908-045f5fab27d8)

- Another version of the step after freeze dont start back the sim. 
- Let the sim stay in freeze.
- Websocket client doesnot receive messages for a while(but not much as mentioned steps above) 
- 16 seconds to receive messages again. Check the figure below:
![image](https://github.com/nasa/trick/assets/40665614/898e89c3-3234-40bf-a8eb-1f0376113489)

**Comment by sharmeye** [comment]
The webserver functionality is not very well-supported for a variety of reasons, but we will look into this and determine what is causing the holdup.  Thanks for bringing it to our attention.

**Comment by Fjolnirr** [comment]
Do the web socket and socket variable servers use the same approach to gather variables from main sim thread?
 I haven't looked it up but aren't they using the same layer to collect data from sim?

**Comment by sharmeye** [comment]
They should use the same underlying functionality now (in fact this comment made me find an old issue that should've been closed like 2 years ago about this).  We've identified the problem, and we're working on a fix right now.

**Comment by Mrockwell2** [comment]
@Fjolnirr The linked branch and pull request should fix the issue. Please test it out and let me know if you still run into any problems.

**Comment by Fjolnirr** [comment]
I have installed trick on this branch https://github.com/Mrockwell2/trick/tree/1720_Websocket_Freeze_Issue 
Still got the similar results. 
![image](https://github.com/nasa/trick/assets/40665614/4b9300fc-bbc4-4970-a211-2b10e13727da)

**Comment by Mrockwell2** [comment]
> I have installed trick on this branch https://github.com/Mrockwell2/trick/tree/1720_Websocket_Freeze_Issue Still got the similar results. ![image](https://private-user-images.githubusercontent.com/40665614/336115008-4b9300fc-bbc4-4970-a211-2b10e13727da.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc0MzAxNDEsIm5iZiI6MTcxNzQyOTg0MSwicGF0aCI6Ii80MDY2NTYxNC8zMzYxMTUwMDgtNGI5MzAwZmMtYmJjNC00OTcwLWEyMTEtMmIxMGUxMzcyN2RhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjAzVDE1NTA0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgxNmQxNTk2YWVlNDdhNjY1ZTI1MDJhYWIyZjcxN2Q5ZWJhYjZhMjY0NDQ2MTAzNDhlMmZiZWNkYzFjMDIyY2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QPF8Y31ibqg_KGBGK1zdwlPD1a1v62wHBwx_ZdARFoU)

Did you do a make clean before rebuilding?

**Comment by Fjolnirr** [comment]
Of course. I have cloned your repo and installed everything from 0.

**Comment by hchen99** [comment]
I tried on both Firefox & Safari (on Mac) with `WS Experiments` in `SIM_cannon_numeric` as #1294 suggested without any issue. Without the fix in this PR, I did see the web client didn't get data updates after freeze. However with the fix, the client received data updates after freeze without any issue. 
On the other hand, Chrome doesn't work for me for some reason. Those variables specified in the respective wsexp.html didn't even show up on Chrome.
I didn't try `postman` though.
